File: Java/백준/Gold/1107. 리모컨/리모컨.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        int N = s.nextInt(); //이동하고자 하는 채널
        int M = s.nextInt(); //고장난 버튼의 개수

        int[] button = new int[M]; //고장난 버튼
        for (int i = 0; i < M; i++) {
            button[i] = s.nextInt();
        }

        int minClicks = Math.abs(100 - N);

        for (int i = 0; i < 1000001; i++) { //채널은 무한대 만큼 있음
            if (validChannel(i, button)) {
                int clicks = String.valueOf(i).length() + Math.abs(N - i);
                minClicks = Math.min(minClicks, clicks);
            }
        }

        System.out.println(minClicks);
    }

    public static boolean validChannel(int N, int[] button) {
        String nString = String.valueOf(N);
        for (int i = 0; i < nString.length(); i++) {
            int digit = nString.charAt(i) - '0';
            for (int j = 0; j < button.length; j++) {
                if (digit == button[j]) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: C11/백준/Gold/1107. 리모컨/리모컨.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

int min(int a, int b) {
    return a < b ? a : b;
}

bool validChannel(int N, int M, int button[]) {
    char buffer[10];
    sprintf(buffer, "%d", N);
    for (int i = 0; i < strlen(buffer); i++) {
        int digit = buffer[i] - '0';
        for (int j = 0; j < M; j++) {
            if (digit == button[j]) {
                return false;
            }
        }
    }
    return true;
}

int main(void) {
    int N = 0;
    int M = 0;
    scanf("%d %d", &N, &M);
    
    int button[M];
    for (int i = 0; i < M; i++) {
        scanf("%d", &button[i]);
    }
    
    int minClicks = abs(100 - N);
    
    for (int i = 0; i < 1000001; i++) {
        if (validChannel(i, M, button)) {
            char buffer[10];
            sprintf(buffer, "%d", i);
            int clicks = strlen(buffer) + abs(N - i);
            minClicks = min(minClicks, clicks);
        }
    }
    
    printf("%d", minClicks);
    
    return 0;
}